packages/firestore/scripts/assertion-id-tool.ts

```
@@ -0,0 +1,372 @@
+            functionName = expression.text;
+          }
+
+          // If we found a function name, and it's one we're looking for
+          if (functionName && targetFunctionNames.has(functionName)) {
+            // Get line and character number
+            const { line, character } = ts.getLineAndCharacterOfPosition(
+              sourceFile,
+              node.getStart() // Get start position of the call expression
+            );
+
+            // --- Extract Arguments ---
+            const argsText: string[] = [];
+            let errorMessage: string | undefined;
+            let assertionId: string | undefined;
+            if (node.arguments && node.arguments.length > 0) {
+              node.arguments.forEach((arg: ts.Expression) => {
+                // Get the source text of the argument node
+                argsText.push(arg.getText(sourceFile));
+
+                if (ts.isStringLiteral(arg)) {
+                  errorMessage = arg.getText(sourceFile);
+                }
+                else if (ts.isNumericLiteral(arg)) {
+                  assertionId = arg.getText(sourceFile);
+                }
+              });
+            }
+            // --- End Extract Arguments ---
+
+            // Store the information (add 1 to line/char for 1-based indexing)
+            foundCalls.push({
+              fileName: filePath, // Store the full path
+              functionName,
+              line: line + 1,
+              character: character + 1,
+              argumentsText: argsText, // Store the extracted arguments,
+              errorMessage,
+              assertionId: assertionId ?? "INVALID",
+            });
+          }
+        }
+
+        // Continue traversing down the AST
+        ts.forEachChild(node, visit);
+      };
+
+      // Start traversal from the root SourceFile node
+      visit(sourceFile);
+
+    } catch (error: any) {
+      console.error(`Error processing file ${filePath}: ${error.message}`);
```

Again, strongly consider just letting the exception bubble up and terminate the app. This "should" never fail, and if it does, IMO it should fail loudly.

---

Good call. Removed the try catch.